perma_web/frontend/components/UploadForm.vue

```diff
@@ -63,57 +46,61 @@ const handleClick = (e) => {
 }
 
 const handleUploadRequest = async () => {
-    if (!globalStore.isReady) {
+    if (captureStatus.value !== 'ready') {
         return
     }
 
-    handleErrorReset()
-    globalStore.captureStatus = "isValidating"
-
-    const csrf = getCookie("csrftoken")
-    const requestType = globalStore.captureGUID ? "PATCH" : "POST"
-    const requestUrl = "/archives/" + globalStore.captureGUID ? `${globalStore.captureGUID}/` : ""
+    // validation
+    errors.value = {}
+    if (!file.value) {
+        errors.value.file = ['File is required']
+    }
+    if (!props.captureGUID) {
+        if (!title.value) {
+            errors.value.title = ['Title is required']
+        }
+        if (!url.value) {
+            errors.value.url = ['URL is required']
+        }
+    }
+    if (Object.keys(errors.value).length > 0) {
+        return
+    }
 
+    // prepare form data
+    captureStatus.value = "isValidating"
     const formDataObj = new FormData();
     formDataObj.append('folder', globalStore.selectedFolder.folderId);
-
-    for (const [key, { value }] of Object.entries(formData.value)) {
-        formDataObj.append(key, value);
+    formDataObj.append('file', file.value);
+    if (!props.captureGUID) {
```

Comment:
Just noting that the non-Vue version of this form actually doesn't do the `patch` version of this: it always makes a new Perma Link.

So, we'll have to see if no title/description/url, ends up being what users want here. Could end up that they prefer the current behavior.

Reply:
Yeah huh, I didn't actually trigger that version in my testing, just did captures of invalid domains. How does one show it?

Reply:
I think the quickest way is, you could throw a `1/0` or `raise HaltCaptureException` someplace in https://github.com/harvard-lil/perma/blob/develop/perma_web/perma/celery_tasks.py#L312, to make a failed capture.